modules/planning/common/reference_line_info.cc

```diff
@@ -219,15 +219,22 @@ bool ReferenceLineInfo::GetFirstOverlap(
   constexpr double kMaxOverlapRange = 500.0;
   double overlap_min_s = kMaxOverlapRange;
 
-  for (const auto& overlap : path_overlaps) {
-    if (overlap.end_s < start_s) {
+  auto overlap_min_s_iter = path_overlaps.end();
+  for (auto iter = path_overlaps.begin(); iter != path_overlaps.end(); ++iter) {
+    if (iter->end_s < start_s) {
       continue;
     }
-    if (overlap_min_s > overlap.start_s) {
-      *path_overlap = overlap;
-      overlap_min_s = overlap.start_s;
+    if (overlap_min_s > iter->start_s) {
+      overlap_min_s_iter = iter;
+      overlap_min_s = iter->start_s;
     }
   }
+
+  // Ensure that the path_overlaps is not empty.
+  if (overlap_min_s_iter != path_overlaps.end()) {
+    *path_overlap = *overlap_min_s_iter;
+  }
+
```

Comment:
If the purpose of this change is just to avoid multiple "copy" action in line 227, you can just create a local pointer (hdmap::PathOverlap*), and set it to &overlap at line 227, and then at the end, copy it to *path_overlap if not nullptr.

no need to change so many lines of code.

Reply:
@jmtao Thanks. First, I prefer to use iterators so that the C++ STL style is more obvious. Second,

Reply:
Thanks once more for your comment.